source/documentation/deploying-an-app/using-circleci-for-continuous-deployment.html.md.erb

```diff
@@ -0,0 +1,262 @@
+
+```yaml
+k get secret <NAME-OF-CircleCI-SERVICE-ACCOUNT-SECRET> -o yaml -n <NAMESPACE | grep ca.crt
+```
+
+Likewise grab the CircleCI service accont token as follows: 
+
+```yaml
+k get secret <NAME-OF-CircleCI-SERVICE-ACCOUNT-SECRET> -o yaml -n <NAMESPACE | grep token
+
+The token will need to be decoded as follows: 
+
+echo <ENCODED_TOKEN> | base 64 --decode
+```
+
+Both the ca.crt and token will be added as environment variables by creating a context in the CircleCI console, that holds these values as shown below. 
+
+
+### Linking Repository to CircleCI
+MoJ has as an account with CircleCI, please login to [CircleCI](https://circleci.com) using GitHub credentials. Select project, and if config.yml is in the repo CircleCI will build and run tests.
+
+### Add variables to CircleCI
+There is a number of environment variables that you will need to set on your CircleCI project in order to build a docker image, push it to the ECR and trigger a deployment in your environment. On the project page, click the cog icon in the left pane. Click on 'Contexts'. Click 'Create Context'.
+Enter your preferred name for your context. Would make sense to add the environment as prefix to the context, i.e dev or prod.
+The context will hold your environment variables and later be used by the docker executor of the pipeline. 
+
+Add the following environment variables:
+
+#### AWS credentials
+
+To authenticate with ECR, you will need to set:
+
+- `AWS_DEFAULT_REGION` - would be `eu-west-2` for Cloud Platform clusters unless specified otherwise
+- `AWS_ACCESS_KEY_ID`
+- `AWS_SECRET_ACCESS_KEY`
+- `ECR_ENDPOINT` is optional but useful if you want to avoid having to hardcode the full hostname of the registry
+
+#### Kubernetes `ServiceAccount` credentials
+Since a single CircleCI project will need to access multiple namespaces in kubernetes (the environments), it will also need to handle multiple credentials. To simplify authentication, we provide a helper script in our supported [build image](https://github.com/ministryofjustice/cloud-platform-tools-image). For a usage example, see [Deploy To Kubernetes](/documentation/deploying-an-app/use-circleci-to-upgrade-app.html#upload-to-ecr) below.
+
+There are four different variables that CircleCI will need to access the cluster: 
+
+- `K8S_CLUSTER_NAME` - the full name of the cluster (eg.: `live-1.cloud-platform.service.justice.gov.uk`)
+- `K8S_NAMESPACE` - the name of the `Namespace` (see [Create a namespace][env-create])
+- `K8S_CLUSTER_CERT` - the CA Certificate for the cluster, can be acquired from the `Secret` that is generated for the `ServiceAccount`
+- `K8S_TOKEN` - the access token generated for the `ServiceAccount`. Please note, you should first base64 decode the token value you retrieve from the secret [in the previous section](/documentation/deploying-an-app/use-circleci-to-upgrade-app.html#creating-a-service-account-for-circleci), e.g. `echo <thereallylongstringthatyougetback> | base64 --decode`.
```

Comment:
The "in the previous section" link is broken.

We generally use link aliases at the foot of the page for any links which appear more than once in a document. i.e.

```
This is [my link][alias], and [here][alias] it is again.
...
[alias]: https://google.com
```

Reply:
done.